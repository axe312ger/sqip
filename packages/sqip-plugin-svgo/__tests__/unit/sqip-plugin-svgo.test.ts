import SVGO from 'svgo'
import SvgoPlugin from '../../src/sqip-plugin-svgo'

jest.mock('svgo')

const optimizeMock = jest.fn(() => Promise.resolve({ data: 'mocked' }))

SVGO.mockImplementation(function SVGOmock() {
  this.optimize = optimizeMock
})

test('runSVGO', async () => {
  const svgoPlugin = new SvgoPlugin({})
  const inputSVG = Buffer.from('<svg />')
  await svgoPlugin.apply(inputSVG)
  expect(SVGO).toHaveBeenCalledTimes(1)
  expect(SVGO.mock.calls[0]).toMatchSnapshot()
  expect(optimizeMock).toHaveBeenCalledTimes(1)
  expect(optimizeMock.mock.calls[0][0]).toBe(inputSVG)
})
