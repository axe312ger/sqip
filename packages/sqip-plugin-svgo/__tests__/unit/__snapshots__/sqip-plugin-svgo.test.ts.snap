// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runSVGO 1`] = `
Array [
  Object {
    "floatPrecision": 1,
    "multipass": true,
    "plugins": Array [
      Object {
        "removeDoctype": true,
      },
      Object {
        "removeXMLProcInst": true,
      },
      Object {
        "removeComments": true,
      },
      Object {
        "removeMetadata": true,
      },
      Object {
        "removeXMLNS": false,
      },
      Object {
        "removeEditorsNSData": true,
      },
      Object {
        "cleanupAttrs": true,
      },
      Object {
        "inlineStyles": true,
      },
      Object {
        "minifyStyles": true,
      },
      Object {
        "convertStyleToAttrs": true,
      },
      Object {
        "cleanupIDs": true,
      },
      Object {
        "prefixIds": true,
      },
      Object {
        "removeRasterImages": true,
      },
      Object {
        "removeUselessDefs": true,
      },
      Object {
        "cleanupNumericValues": true,
      },
      Object {
        "cleanupListOfValues": true,
      },
      Object {
        "convertColors": true,
      },
      Object {
        "removeUnknownsAndDefaults": true,
      },
      Object {
        "removeNonInheritableGroupAttrs": true,
      },
      Object {
        "removeUselessStrokeAndFill": true,
      },
      Object {
        "removeViewBox": false,
      },
      Object {
        "cleanupEnableBackground": true,
      },
      Object {
        "removeHiddenElems": true,
      },
      Object {
        "removeEmptyText": true,
      },
      Object {
        "convertShapeToPath": true,
      },
      Object {
        "moveElemsAttrsToGroup": true,
      },
      Object {
        "moveGroupAttrsToElems": true,
      },
      Object {
        "collapseGroups": true,
      },
      Object {
        "convertPathData": true,
      },
      Object {
        "convertTransform": true,
      },
      Object {
        "removeEmptyAttrs": true,
      },
      Object {
        "removeEmptyContainers": true,
      },
      Object {
        "mergePaths": true,
      },
      Object {
        "removeUnusedNS": true,
      },
      Object {
        "sortAttrs": true,
      },
      Object {
        "removeTitle": true,
      },
      Object {
        "removeDesc": true,
      },
      Object {
        "removeDimensions": true,
      },
      Object {
        "removeAttrs": false,
      },
      Object {
        "removeAttributesBySelector": false,
      },
      Object {
        "removeElementsByAttr": false,
      },
      Object {
        "addClassesToSVGElement": false,
      },
      Object {
        "removeStyleElement": false,
      },
      Object {
        "removeScriptElement": false,
      },
      Object {
        "addAttributesToSVGElement": false,
      },
      Object {
        "removeOffCanvasPaths": true,
      },
      Object {
        "reusePaths": true,
      },
    ],
  },
]
`;
